### DISCUSSION TITLE
Creating Kuberenetes pods along with namespaces

### DISCUSSION THREAD

**Discussion by UmamaheshMaxwell** [BretFisher/ama#227] [Q&A]
Hey Bret, 

Can I create namespace dynamically while creating a pod using kubectl run, below is my command I know it wouldn't work, but I am curious to know is there any way to create namespace using Kubectl run statement ? or is it mandatory for me create namespace in advance and then use run statement to use the created namespace.   

``` kubectl run webproxy --image nginx --port 8080 -l "env=dev" -n webproxy ```

Error :  ``` Error from server (NotFound): namespaces "webproxy" not found```

**Comment by BretFisher** [comment]
Sorry, I don't know of a way to do that. If you need a single command to create both a pod and namespace, then use `kubectl apply -f` with a file that has the pod and namespace in it.
